### PR TITLE
Direct Link Ports - Data Source Set 

### PR DESCRIPTION
**Pull request by ibmterraformdl**

### CODE REVIEW

File: ibm/data_source_ibm_dl_port.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+func dataSourceIBMDirectLinkPortRead(d *schema.ResourceData, meta interface{}) error {
+
+	sess, err := meta.(ClientSession).DirectlinkV1API()
+	if err != nil {
+		return err
+	}
+
+	getPortsOptions := sess.NewGetPortOptions(d.Get(dlPortID).(string))
+	response, _, err := sess.GetPort(getPortsOptions)
+	if err != nil {
```
Comment (hkantare) [inline ibm/data_source_ibm_dl_port.go:62]: When we try with an invalid port ID does this give any error?

File: ibm/data_source_ibm_dl_port.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			},
+			dlLocationName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Port location long name",
+			},
+			dlProviderName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Port location long name",
```
Comment (hkantare) [inline ibm/data_source_ibm_dl_port.go:39]: description is same for all dlLocationDisplayName, dlLocationName, dlLocationName

File: ibm/data_source_ibm_dl_port_test.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+func TestAccIBMDLPortDataSource_basic(t *testing.T) {
+	name := "dl_port"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMDLPortDataSourceConfig(name),
+				Check:  resource.ComposeTestCheckFunc(),
```
Comment (hkantare) [inline ibm/data_source_ibm_dl_port_test.go:18]: you can add some attributes set checks 
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/data_source_ibm_tg_gateways_test.go

File: website/docs/d/dl_port.html.markdown [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+## Attribute Reference
+
+The following attributes are exported:
+
+* `direct_link_count` - Count of existing Direct Link gateways in this account on this port.
+* `label` - Port Label.
+* `location_display_name` - Port location long name.
+* `location_name` - Port location name identifier.
+* `port_id` - Port identifier.
```
Comment (hkantare) [inline website/docs/d/dl_port.html.markdown:36]: Remove port_id from attribute already part of argument

File: ibm/data_source_ibm_dl_port.go (hunk 1 above)
Comment (ibmterraformdl) [reply ibm/data_source_ibm_dl_port.go:62]: Yes. It gives, invalid ID error.

**Pull request merged**